### ISSUE TITLE
LCD backlight is flickering when heaters are active

### ISSUE THREAD

**Issue by creamywhite** [bigtreetech/BIGTREETECH-SKR-mini-E3-#24]
I have a stock Ender 3 display and it's backlight flickers (changes intensity) slightly if hotend or heatbed is powered. I suspect it's a hardware issue - is there any known fix for this?

**Comment by hopatz** [comment]
Same problems here

**Comment by creamywhite** [comment]
I was able to solve the issue by soldering 4.7 uF capacitor across +5V and GND pins of EXP1 connector. Seems that the board has some electric design flaws after all. My BLTouch (which worked flawlessly before on a stock Ender 3 board) also would randomly deploy its pin during prints until I added 100 nF decoupling capacitor across PA1 anf GND pins of SERVO connector.

**Comment by sky8000** [comment]
Hi, that happened to me too.

Check if you have PID enabled for hot bed and disable it (comment #define PIDTEMPBED ). This should also fix your problem.

**Comment by creamywhite** [comment]
> Hi, that happened to me too.
> 
> Check if you have PID enabled for hot bed and disable it (comment` #define PIDTEMPBED` in **Configuration.h**). This should also fix your problem.

Thanks for the suggestion. However, disabling a useful feature (which bed PID is) to workaround a possible electrical design issue seems wrong to me.

**Comment by sky8000** [comment]
> 
> 
> > Hi, that happened to me too.
> > Check if you have PID enabled for hot bed and disable it (comment` #define PIDTEMPBED` in **Configuration.h**). This should also fix your problem.
> 
> Thanks for the suggestion. However, disabling a useful feature (which bed PID is) to workaround a possible electrical design issue seems wrong to me.

I agree that it is an useful feature but not a requirement for the heated bed where there's no need for very high precision temperatures. Anyway I noticed that Bigtreetech has an external heated bed power module called BTMOS v2.0 that supports current up to 15A (https://www.aliexpress.com/item/32874506353.html). I wonder if the module would solve this problem.

**Comment by creamywhite** [comment]
> > > Hi, that happened to me too.
> > > Check if you have PID enabled for hot bed and disable it (comment` #define PIDTEMPBED` in **Configuration.h**). This should also fix your problem.
> > 
> > 
> > Thanks for the suggestion. However, disabling a useful feature (which bed PID is) to workaround a possible electrical design issue seems wrong to me.
> 
> I agree that it is an useful feature but not a requirement for the heated bed where there's no need for very high precision temperatures. Anyway I noticed that Bigtreetech has an external heated bed power module called BTMOS v2.0 that supports current up to 15A (https://www.aliexpress.com/item/32874506353.html). I wonder if the module would solve this problem.

First of all, it is debatable whether heatbed temperature stability is necessary or not. Secondly, using bang-bang heating method puts more load on electrical components. Yes, the MOSFET that is used for bed heating control (IPD036N04L) is rated well above 10A that Ender 3's heatbed can draw, so there shouldn't be any issues. I'm just a bit frustrated that SKR Mini E3 is designed badly in some aspects causing users to seek workarounds if they want to use proven useful features.

**Comment by JimmySnails** [comment]
@creamywhite i noticed the same thing but without PWM. `#define PIDTEMPBED` is disabled per default and still my LCD brightness jumps every now and then. It doesn't happen too often but it's annoying anyway and i'd like to fix that.

So i'd like to try adding capacitor like you did. I'm just confused you said you added it on EXP1? My LCD is connected via EXP3 (like on the stock board). So did you add the capacitor between the empty pins on the LCD screen? or on the ribbon cable? If you used the ribbon cable, which colors are  +5V and GND ?

Thank you!

**Comment by creamywhite** [comment]
> @creamywhite i noticed the same thing but without PWM. `#define PIDTEMPBED` is disabled per default and still my LCD brightness jumps every now and then. It doesn't happen too often but it's annoying anyway and i'd like to fix that.
> 
> So i'd like to try adding capacitor like you did. I'm just confused you said you added it on EXP1? My LCD is connected via EXP3 (like on the stock board). So did you add the capacitor between the empty pins on the LCD screen? or on the ribbon cable? If you used the ribbon cable, which colors are +5V and GND ?
> 
> Thank you!

Sorry for a late answer. I soldered a capacitor directly between 5V and GND pins of EXP1 connector on the board. You can refer to this diagram to locate them https://cdn.shopify.com/s/files/1/1619/4791/files/SKR-mini-E3_05.jpg?v=1562825161. Also you can solder a capacitor of a higher value than 4.7 uF. As I noticed it won't solve the issue completely, but the screen will flicker less.

**Comment by kirelam** [comment]
Hi,

I am considering to buy the SKR mini E3 for a while now. 

But i read that there seems to be an issue with the 5V voltage regulator. It seems to leak 1.5V to the USB port and it doesn't seem to be stable, where it causes strange beheviour of some components, like the BLtouch. Is there a new revision planned, or is there a modification known here? i am willing to do this, but i do not know what i have the change or add to the board. I have some Solder Skills, so that is no problem!

Many regards,
Kirelam

**Comment by barko99** [comment]
Hey there,

even this discussion is a bit old, i want to comment my issues with the skr regarding EMI:

After fiddling around with uneven extrusion over height, (changing nearly everything regarding z axis) i switched over to check if my uneven extrusion is in just one layer. Same here.  More or less periodically, i have a lowered extrusion and z axis. I changed everything regarding extrusion (motor, extruder, bowden, nozzle, etc. etc.)

After a lot of head scratching, i noticed that i have these issues when the heatbed is heating. It is reproducible, so while printing, i can change to a higher bed temp, and suddenly i have the issues with lower extrusion feedrate and my z axis drops down slightly. 

I do not have solved the issue right now, i want to try an external mosfet the following days and switching off the PID bed.

Best regards and wishes

Johannes

**Comment by txt4nk** [comment]
Did anyone ever completely solve this?

**Comment by barko99** [comment]
Honestly, i can't really help you with display flickering. My above issue was caused (or exaggerated) by my swap heatbed. 

I had an Fsytec magnetic heatbed installed. I changed nearly everything while trying to find the issue. (external mosfet, external mosfet for heatbed and hotend, new PSU, second new PSU (one for heatbed, one for hotend and SKR, .... .... .... )

The last thing i thought of was the heatbed. I couldn't see why and how this could effect my SKR. Still, while not knowing what the issue is with this bed i swaped to the stock bed again. The stock bed and SKR works now perfectly in my case.

Regards

**Comment by txt4nk** [comment]
When I had my E3 mini, I was on an oem heat bed. After the E3 mini fried, I installed the th3d ezboard lite along with a spool3d forged aluminum heat bed with integrated magnets (amazing bed, btw!) And i never once had this issue. As soon as I put the E3 DIP in the printer, the problem returned. I have a meanwell psu on the way to replace the landy, along with a bunch of ferrite chokes. If that doesn't work ill end up soldering a resistor on to EXP1 and may even rewire my printer with all shielded cables.. kind of ridiculous though, I guess thats why the ezboard costs so much..

**Comment by txt4nk** [comment]
So I got a response to the issue I created about this that I have yet to see, and could possibly help..
https://github.com/bigtreetech/BIGTREETECH-SKR-E3-DIP-V1.0/issues/72#issuecomment-628454817

**Comment by txt4nk** [comment]
well, that comment I linked to 100% worked. I've had the screen sitting to the side of the printer, not touching the frame at all and the screen has not spazzed out even once. It happened on every single print before this. I am designing a sturdy buffer\stand off to prevent metals from touching so I can re-attach it to the frame without EMI issues.

**Comment by txt4nk** [comment]
Here's the buffer\stand off I created. It works perfectly, I have not had a single glitch since installing this. It's currently 3mm thick, it's sturdy enough but it does have a little more flex than i'd like so I'm going to revise it and go to 6mm. I can provide the STL for whoever has an Ender 5.

![20200516_153342](https://user-images.githubusercontent.com/65292849/82130078-aacd8500-978d-11ea-95df-5652a9dd608a.jpg)
![20200516_153336](https://user-images.githubusercontent.com/65292849/82130079-ac974880-978d-11ea-8abb-42d54c2db45f.jpg)

**Comment by paul-1** [comment]
I think most just remove the screen, and protect the edge of the screen with electrical tape.

**Comment by creamywhite** [comment]
@txt4nk, it nice that you've found a solution to your issue, but I'm afraid it has nothing to do with my issue (and this topic respectively) in terms of its cause and solution. The thing is, in my case, the LCD screen's backlight will flicker slightly when heatbed power is regulated through PWM even when the LCD screen board is not attached to anything (save for EXP cable).

**Comment by txt4nk** [comment]
Ah.. I guess I misunderstood the issue. Still seems to be some sort of EMI in both cases. Have you tried ferrite rings or chokes? Could also try shielded cables.

**Comment by creamywhite** [comment]
> Ah.. I guess I misunderstood the issue. Still seems to be some sort of EMI in both cases. Have you tried ferrite rings or chokes? Could also try shielded cables.

I did not and I believe they won't help (might try just out of curiosity if I find one). As I stated in my previous messages here, the issue is mitigated to a degree by a capacitor across VCC and GND pins.

**Comment by txt4nk** [comment]
I'd throw some chokes on the PSU to main board wires, main board to bed and main board to hot end.  IF the issue is something a choke would stop, those three points would stop it regardless of the source, whether it be the PSU or the main board. Chokes are super cheap, so it's worth trying at least.
